drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -855,6 +856,37 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
 }
 #endif /* CONFIG_NRF_802154_ENCRYPTION */
 
+#if defined(CONFIG_IEEE802154_CSL_ENDPOINT)
+
+static void nrf5_receive_at(const struct ieee802154_config *config)
+{
+	bool primary_cancel = nrf_802154_receive_at_scheduled_cancel(DRX_SLOT_RX_PRIMARY);
+	bool secondary_cancel = nrf_802154_receive_at_scheduled_cancel(DRX_SLOT_RX_SECONDARY);
+	if (!primary_cancel && !secondary_cancel) {
+		/* Edge case: In between the primary and secondary slot cancellation
+		 *            the secondary RX slot might have started. Due to ordering
+		 *            of the above two cancellations we can assume that the primary
+		 *            slot is already free.
+		 */
+		primary_cancel = true;
+	}
+
+	int slot_id = primary_cancel ? DRX_SLOT_RX_PRIMARY : DRX_SLOT_RX_SECONDARY;
```

Comment:
Please look at prototype:
```c
bool nrf_802154_receive_at(uint64_t rx_time,
                           uint32_t timeout,
                           uint8_t  channel,
                           uint32_t id);
```
The `slot_id` should be of `uint32_t` type to avoid implicit conversion from `int` to `uint32_t`
```suggestion
	uint32_t slot_id = primary_cancel ? DRX_SLOT_RX_PRIMARY : DRX_SLOT_RX_SECONDARY;
```

Reply:
Changed to `uint32_t`